### DISCUSSION TITLE
[Bug]: [MM2] Some partitions stop being replicated after rebalancing

### DISCUSSION THREAD

**Discussion by dzasta997** [strimzi/strimzi-kafka-operator#10613] [General]
### Bug Description

Some time ago we spotted MirrorMaker not replicating some partitions after rebalancing. Logs below are produced by some worker but manual restarting it not always helps, sometimes other pods also require to be restarted, not necessarily all of them.

It started first time few days after updating operator to `0.42.0`, but it lasts in `0.43.0`.

![image](https://github.com/user-attachments/assets/b8c11359-6a51-46bc-b436-e3f7b827997a)
![image](https://github.com/user-attachments/assets/df731a35-af66-4a9a-9ee4-a6401d565ce6)

We didn't change MM configuration after updates.

### Steps to reproduce

Only happens after rebalancing

### Expected behavior

All partitions are replicated or worker automatically restarts the process in case of such error

### Strimzi version

0.43.0

### Kubernetes version

1.29

### Installation method

Helm chart

### Configuration files and logs

> 2024-09-19 08:04:36,737 ERROR [kafka>kafka.MirrorSourceConnector|worker] Connector 'kafka>kafka.MirrorSourceConnector' failed to properly shut down, has become unresponsive, and may be consuming external resources. Correct the configuration for this connector or remove the connector. After fixing the connector, it may be necessary to restart this worker to release any consumed resources. (org.apache.kafka.connect.runtime.Worker) [StartAndStopExecutor-connect-mirror-maker2-mirrormaker2-2.mirror-maker2-mirrormaker2.kafka.svc:8083-3]

**Comment by scholzj** [comment]
Please keep in mind that Mirror Maker 2 is a Apache Kafka project. Strimzi only operates it. Also, sharing a single log line without even explaining properly where it comes from is pretty useless. You should share the full logs and configurations (custom resources).
